src/main/java/com/aws/greengrass/shadowmanager/configuration/ComponentConfiguration.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.shadowmanager.configuration;
+
+import com.aws.greengrass.config.Topics;
+
+import static com.aws.greengrass.componentmanager.KernelConfigResolver.CONFIGURATION_CONFIG_KEY;
+
+public final class ComponentConfiguration {
+    RateLimitsConfiguration rateLimitsConfiguration;
+
+    private ComponentConfiguration(RateLimitsConfiguration rateLimitsConfiguration) {
+        this.rateLimitsConfiguration = rateLimitsConfiguration;
+    }
+
+    /**
+     * Constructor for shadow manager component.
+     * @param oldConfiguration previous configuration of the component
+     * @param updatedTopics current configuration topics
+     * @return shadow manager component configuration object
+     */
+    public static ComponentConfiguration from(ComponentConfiguration oldConfiguration, Topics updatedTopics) {
+        Topics serviceTopics = updatedTopics.lookupTopics(CONFIGURATION_CONFIG_KEY);
+        RateLimitsConfiguration rateLimitsConfiguration = RateLimitsConfiguration.from(oldConfiguration, serviceTopics);
```

Comment:
pass the `RateLimitsConfiguration`.
```suggestion
        RateLimitsConfiguration rateLimitsConfiguration = RateLimitsConfiguration.from(getRateLimitsConfiguration(oldConfiguration), serviceTopics);
        
      ....
        
  private static RateLimitsConfiguration getRateLimitsConfiguration(ComponentConfiguration oldConfiguration) {
     if (oldConfiguration != null) {
         return oldConfiguration.getRateLimitsConfig();
    }

    return null;
}
```